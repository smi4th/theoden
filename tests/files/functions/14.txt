fonction functionNAME(n, acc) {
    if (n == 0) {
        return acc;
    } else {
        return functionNAME(n - 1, acc + 1);
    }
}

functionNAME(5, 1);