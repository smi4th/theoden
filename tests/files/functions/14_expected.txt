---TEST OUTPUT---
---MEMORY---
---FUNCTIONS---
functionNAME = (('param', ('param', 'n'), 'acc'), ('linst', ('if', ('==', ('var', 'n'), ('int', 0)), ('linst', ('return', ('var', 'acc'))), ('else', ('linst', ('return', ('call', 'functionNAME', ('expr', ('expr', ('-', ('var', 'n'), ('int', 1))), ('+', ('var', 'acc'), ('int', 1))))))))))
